libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -96,6 +96,7 @@ internal class MapboxTripSession(
                 )
             }
             is SetRefreshedRoutesInfo -> {
+                isUpdatingRoute = true
```

Comment:
@LukasPaczos , @RingerJK , @dzinad , do you think we need to ignore instructions while NN parces alternatives? I would keep `isUpdatingRoute = true` only for `BasicSetRoutesInfo`

Reply:
I'm wondering why do we stop updates at all? I guess we don't want to receive status updates which are related to the previous route as soon as we set a new route, do we?
If that so, maybe it's better to implement in on NN side: navigator should stop status updates as soon as new route is set? What do you think?

Reply:
I don't know why we stop the updates but I can suppose that we don't want to trigger `RoutesObserver` and other observers with data for an old route if new routes have already been set (the API method was invoked so from the user's point of view they were already set).

>  I would keep isUpdatingRoute = true only for BasicSetRoutesInfo

Why?

Reply:
UPD: I guess this PR does not break what I described above because alternative refresh does not affect primary route. Not sure about other reasons though.

Reply:
`isUpdatingRoute` prevents `RouteProgressObserver` from getting updates, not `RoutesObserver`.

> Why?

It seems okay for me to get route progress triggered by location update while NN refreshes the route or parses new alternatives. What do you think about it?

Reply:
The goal for the cutoff is that there are no new `RouteProgress` updates dispatched with old route reference after developer calls `MapboxNavigation#setNavigationRoutes`, as this could be considered surprising.

Given this requirement, I think that only pausing `RouteProgress` updates while primary route is set is good enough, we don't need to withhold the updates for alternatives or refresh updates. Alternatives can be set by developers but they do not impact the primary route so it's safe to keep going. Refreshed route cannot be provided by a developer so they are not even aware that it's coming, so it's also safe to keep going.

Reply:
> It seems okay for me to get route progress triggered by location update while NN refreshes the route or parses new alternatives. What do you think about it?

But route refresh impacts the primary route. If we only change the flag for BasicSetRouteInfo, the following case is possible:
1. `RouteRefreshStateObserver` receives REFRESH_STATE_FINISHED_SUCCESS;
2. `RouteProgressObserver` receives old route related progress.

Reply:
We don't notify `RouteRefreshStateObserver`s with the refreshed routes, so I don't think there is a problem from developer's point of view. Developers get the new routes via `RoutesObserver`s, and they are invoked only when the routes have been actually set to NavNative.